File: Models/FurthereducationInProgress.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Models
{
    public class FurthereducationInProgress
    {
        public int Id { get; set; }
        [Required]
        public int MonitoringId { get; set; }
        public Monitoring Monitoring { get; set; }
        [Required]
        [StringLength(50)]
        public String Ontraining { get; set; }
        public DateTime? GraduateDate { get; set; }
        public DateTime? DropoutDate { get; set; }
        [Required]
        [StringLength(250)]
        public string Reason { get; set; }
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/FurtherEducationInProgress.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "FurthereducationInProgresses")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FurtherEducationInProgress {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MonitoringId")
    private Monitoring monitoring;

    @NotNull
    @Size(max = 50)
    private String ontraining;

    private LocalDateTime graduateDate;
    private LocalDateTime dropoutDate;

    @NotNull
    @Size(max = 250)
    private String reason;
}
